spyder/plugins/updatemanager/workers.py

```diff
@@ -150,12 +151,18 @@ def start(self):
         else:
             url = pypi_url
 
+        headers = {}
+        token = os.getenv('GITHUB_TOKEN')
+        if url == github_url and token:
+            headers.update(Authorization=f"Bearer {token}")
```

Comment:
Since this is only necessary for CIs, I think it needs to be:

```suggestion
        if running_in_ci():
            token = os.getenv('GITHUB_TOKEN')
            if url == github_url and token:
                headers.update(Authorization=f"Bearer {token}")
```

Reply:
How about this?
```suggestion
        token = os.getenv('GITHUB_TOKEN')
        if running_in_ci() and url == github_url and token:
            headers.update(Authorization=f"Bearer {token}")
```